### PR TITLE
Updates from Way Lab specific implementations

### PR DESCRIPTION
**Pull request by d33bs**
This PR updates and adds files which implemented this template though the Way Lab (via [`gc-assayworks-bucket`](https://github.com/WayScience/gc-assayworks-bucket) and [`gc-durbinlab-bucket`](https://github.com/WayScience/gc-durbinlab-bucket)). I merged updates from those repositories and made relevant changes here to retain the generalized application. In addition, I took some time to make sure dependencies were updated where possible.

I plan to follow up in another PR with updates related to decreasing complexity involved with manually replaced text throughout by implementing [cookiecutter](https://github.com/cookiecutter/cookiecutter) or similar here.

Thank you in advance for your review and thoughts on this!

### CODE REVIEW

**Review by vincerubinetti — APPROVED** [review]
I have no experience with Terraform, so I've just reviewed the text and checked the links. Everything looks good to me. The rclone information looks right from my experience using it.

File: README.md [hunk 1]
```diff
@@ -2,36 +2,89 @@
...
 See below for steps which are required for installation.
 
 1. [Create a repository from this template](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template).
-1. [Clone the repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+
+1. [Clone the repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) to your development environment.
+
 1. Install [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
 
+1. Configure Terraform as necessary to your Google Cloud environment.
```
Comment (falquaddoomi) [inline README.md:44]:
```suggestion
1. Configure Terraform as necessary for your Google Cloud environment.
```

File: README.md [hunk 2]
```diff
@@ -2,36 +2,89 @@
...
+
 ## :books:Tutorial
 
-See below for a brief tutorial on how to implement the work found in this repository for your needs.
+See below for brief tutorials on how to implement the work found in this repository for your needs.
+
+### 🎛️ State Management
+
+These steps cover how to store [Terraform state management](https://developer.hashicorp.com/terraform/language/state) files in association with the bucket infrastructure below. Terraform tracks cloud resources it creates as a statefile (`.tfstate`). If multiple people want to manage the same resources at the same time, they all need to have access to the same statefile or else they overwriting or corrupting state data. One option for sharing this statefile is to use a Google Cloud Bucket, which is the option used here.
```
Comment (falquaddoomi) [inline README.md:52]:
```suggestion
These steps cover how to store [Terraform state management](https://developer.hashicorp.com/terraform/language/state) files in association with the bucket infrastructure below. Terraform tracks cloud resources it creates as a statefile (`.tfstate`). If multiple people want to manage the same resources at the same time, they all need to have access to the same statefile or else they risk overwriting or corrupting state data. One option for sharing this statefile is to use a Google Cloud Bucket, which is the option used here.
```

File: README.md [hunk 3]
```diff
@@ -2,36 +2,89 @@
...
+1. Fill in [terraform/state-management/variables.tf](terraform/variables.tf) with values that make sense for your initiative.
+1. Terraform __init__: to ensure Terraform is initialized use command `terraform -chdir=terraform/state-management init`.
+1. Terraform __plan__: to plan the work and observe any needs use command `terraform -chdir=terraform/state-management plan` .
+1. Terraform __apply__: to apply the work and create resources use command `terraform -chdir=terraform/state-management apply`
+
+### 🏗️ Bucket Infrastructure
+
+These steps cover how to control the infrastructure found within this repository.
+
+| <span style="text-align:left;float:left;font-weight:normal;">:exclamation:  Please note: after applying the Terraform code with the steps below, a `service-account.json` file is added to your local directory which contains sensitive data which may enable access to your cloud resources. __This file should not be checked into source control!__</span>   |
```
Comment (falquaddoomi) [inline README.md:66]: It'd be nice to know where the file will be created, specifically; I wasn't sure what "your local directory" meant. Also, you might consider adding the service key to your `.gitignore` to reduce the odds of an accidental commit.

**Review by falquaddoomi — COMMENTED** [review]
Nice work! I left a few comments, but they're all just text suggestions. The code itself looks good as-is.

File: README.md (hunk 1 above)
Comment (d33bs) [reply README.md:44]: Thanks for this!

File: README.md (hunk 2 above)
Comment (d33bs) [reply README.md:52]: Thanks!

File: README.md (hunk 3 above)
Comment (d33bs) [reply README.md:66]: Thank you, great points! I'll add these to further commits.

**Comment by d33bs** [comment]
Thank you @vincerubinetti and @falquaddoomi for your reviews! I've addressed the comments and will now merge.

**Pull request merged**